### PR TITLE
[TTAHUB-2564] Migration, historical TR goals for completed sessions

### PR DESCRIPTION
**Pull request by nvms**
## How to test

- Sanity check the logic.
- If it looks good, running the migration against recent prod data should result in 57~ new goals.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-2564

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/migrations/20240321002253-create-goals-for-completed-sessions.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            if (!eventReportPilot) {
+              // eslint-disable-next-line no-continue
+              continue;
+            }
+
+            const { goal } = eventReportPilot.data;
+
+            const [[newGoal]] = await queryInterface.sequelize.query(
+              `INSERT INTO "Goals" (name, "grantId", "createdAt", "updatedAt", status, "createdVia", source, "onAR", "onApprovedAR")
+              VALUES ('asdfasdf', '${recipient.value}', NOW(), NOW(), 'In Progress', 'tr', 'Training event', true, true)
```
Comment (thewatermethod) [inline src/migrations/20240321002253-create-goals-for-completed-sessions.js:50]: > VALUES ('asdfasdf'

An acronym perhaps

**Comment by thewatermethod** [comment]
```sql   
    SELECT     
      sr.data->'recipients' recipients,
      sr.id AS session_report_id,
      erpg.id AS event_report_pilot_goal_id
    FROM "SessionReportPilots" sr
    INNER JOIN "EventReportPilots" erp ON sr."eventId" = erp.id
    LEFT JOIN "EventReportPilotGoals" erpg ON sr.id = erpg."sessionId"
    WHERE sr.data->>'status' = 'Complete' ORDER BY event_report_pilot_goal_id DESC;  
 ```
 
 I used the above SQL and see event_report_pilot_goal_id with a null value. Should that still be the case?

**Comment by thewatermethod** [comment]
Also, do we need to worry about GoalCollaborators here?

**Comment by nvms** [comment]
> ```sql
>     SELECT     
>       sr.data->'recipients' recipients,
>       sr.id AS session_report_id,
>       erpg.id AS event_report_pilot_goal_id
>     FROM "SessionReportPilots" sr
>     INNER JOIN "EventReportPilots" erp ON sr."eventId" = erp.id
>     LEFT JOIN "EventReportPilotGoals" erpg ON sr.id = erpg."sessionId"
>     WHERE sr.data->>'status' = 'Complete' ORDER BY event_report_pilot_goal_id DESC;  
> ```
> 
> I used the above SQL and see event_report_pilot_goal_id with a null value. Should that still be the case?

Oh good catch. I added some logic that, when there's already a goal created for the completed session, it ensures there's an entry in EventReportPilotGoals

**Comment by nvms** [comment]
> Also, do we need to worry about GoalCollaborators here?

Ah, yeah, probably. I'll have this done soon

**Comment by nvms** [comment]
Thanks for looking at this already. I think it is ready to be looked at again 🙏 

@thewatermethod

**Review by thewatermethod — APPROVED** [review]

**Review by GarrettEHill — APPROVED** [review]

**Pull request merged**